EncountifyAPI/Controllers/UsersController.cs

```diff
@@ -0,0 +1,283 @@
+            return GetUser(id);
+        }
+
+
+        /// <summary>
+        /// Edit an existing user's username
+        /// </summary>
+        [HttpPut("{id}/Username")]
+        public IEnumerable<User> EditUserName(int id, string username)
+        {
+            using (var connection = new SqlConnection(ConnectionString))
+            {
+                connection.Open();
+                string query = "UPDATE Users SET Username = @username WHERE Id = @id";
+                using SqlCommand command = new SqlCommand(query, connection);
+                command.Parameters.AddWithValue("@id", id);
+                command.Parameters.AddWithValue("@username", username);
+                command.ExecuteNonQuery();
+            }
+            return GetUser(id);
+        }
+
+        /// <summary>
+        /// Edit an existing user's password
+        /// </summary>
+        [HttpPut("{id}/Password")]
+        public IEnumerable<User> EditUserPassword(int id, string password)
+        {
+            using (var connection = new SqlConnection(ConnectionString))
+            {
+                connection.Open();
+                string query = "UPDATE Users SET Password = @password WHERE Id = @id";
+                using SqlCommand command = new SqlCommand(query, connection);
+                command.Parameters.AddWithValue("@id", id);
+                command.Parameters.AddWithValue("@password", password);
+                command.ExecuteNonQuery();
+            }
+            return GetUser(id);
+        }
+
+        /// <summary>
+        /// Edit an existing user's email
+        /// </summary>
+        [HttpPut("{id}/Email")]
+        public IEnumerable<User> EditUserEmail(int id, string email)
+        {
+            using (var connection = new SqlConnection(ConnectionString))
+            {
+                connection.Open();
+                string query = "UPDATE Users SET Email = @email WHERE Id = @id";
+                using SqlCommand command = new SqlCommand(query, connection);
+                command.Parameters.AddWithValue("@id", id);
+                command.Parameters.AddWithValue("@email", email);
+                command.ExecuteNonQuery();
+            }
+            return GetUser(id);
+        }
+
+        /// <summary>
+        /// Edit an existing user's permissions
+        /// </summary>
+        [HttpPut("{id}/IsAdmin")]
+        public IEnumerable<User> EditUserIsAdmin(int id, bool isAdmin)
+        {
+            using (var connection = new SqlConnection(ConnectionString))
+            {
+                connection.Open();
+                string query = "UPDATE Users SET IsAdmin = @isAdmin WHERE Id = @id";
+                using SqlCommand command = new SqlCommand(query, connection);
+                command.Parameters.AddWithValue("@id", id);
+                command.Parameters.AddWithValue("@isAdmin", isAdmin);
+                command.ExecuteNonQuery();
+            }
+            return GetUser(id);
+        }
```

Comment:
Lines similar to 206, 207, 208, 210, 213 seems to be repeating in most of controllers. It might be a good idea to move them to function to avoid repeating